bin/gen-pgbouncer-conf.sh

```diff
@@ -67,22 +80,54 @@ tcp_keepidle = ${PGBOUNCER_TCP_KEEPIDLE:-7200}
 tcp_keepintvl = ${PGBOUNCER_TCP_KEEPINTVL:-75}
 EOFEOF
 
+# add certificates, key to pgbouncer.ini and change sslmode 
+# if all of certs and key are present and also adjust the certificates' formate 
+# from outputs of envrionment variables
+#
+if [ -n "${CLIENT_TLS_KEY_FILE}" ] && [ -n "${CLIENT_TLS_CRT_FILE}" ] && [ -n "${CLIENT_TLS_CA_FILE}" ]
+then 
+  echo -e "-----BEGIN CERTIFICATE-----" > ${PGBOUNCER_DIR}/pgbouncer.crt
+  echo $CLIENT_TLS_CRT_FILE | tr ' ' '\n' | sed '1,2d' | head -n -2 >> ${PGBOUNCER_DIR}/pgbouncer.crt
+  echo -e "-----END CERTIFICATE-----"  >> ${PGBOUNCER_DIR}/pgbouncer.crt
+
+  echo -e "-----BEGIN CERTIFICATE-----" > ${PGBOUNCER_DIR}/pgbouncer_ca.crt
+  echo $CLIENT_TLS_CA_FILE | tr ' ' '\n' | sed '1,2d' | head -n -2 >> ${PGBOUNCER_DIR}/pgbouncer_ca.crt
+  echo -e "-----END CERTIFICATE-----"  >> ${PGBOUNCER_DIR}/pgbouncer_ca.crt
+
+  echo -e "-----BEGIN RSA PRIVATE KEY-----" > ${PGBOUNCER_DIR}/pgbouncer.key
+  echo $CLIENT_TLS_KEY_FILE | tr ' ' '\n' | sed '1,4d' | head -n -4 >> ${PGBOUNCER_DIR}/pgbouncer.key
+  echo -e "-----END RSA PRIVATE KEY-----"  >> ${PGBOUNCER_DIR}/pgbouncer.key
+
+  sed -i '/^server_tls_sslmode =.*/c\
+client_tls_sslmode = require \
+client_tls_protocols = secure \
+client_tls_ciphers =  HIGH:!ADH:!AECDH:!LOW:!EXP:!MD5:!3DES:!SRP:!PSK:@STRENGTH \
+client_tls_key_file = '"${PGBOUNCER_DIR}"'/pgbouncer.key \
+client_tls_cert_file = '"${PGBOUNCER_DIR}"'/pgbouncer.crt \
+client_tls_ca_file = '"${PGBOUNCER_DIR}"'/pgbouncer_ca.crt \
```

Comment:
nitpick, but the `ca` is not pgbouncer-exclusive.

Reply:
yes.there will be a lot of possibilities .  In here, I had considered "if the secrets are presented".